Webhooks/SimpleTexting.ashx

```diff
@@ -0,0 +1,524 @@
+    Object IAsyncResult.AsyncState { get { return _state; } }
+    bool IAsyncResult.CompletedSynchronously { get { return false; } }
+
+    /// <summary>
+    /// Gets a <see cref="T:System.Threading.WaitHandle" /> that is used to wait for an asynchronous operation to complete.
+    /// </summary>
+    /// <exception cref="System.NotImplementedException"></exception>
+    public WaitHandle AsyncWaitHandle
+    {
+        get
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SimpleTextingResponseAsync"/> class.
+    /// </summary>
+    /// <param name="callback">The callback.</param>
+    /// <param name="context">The context.</param>
+    /// <param name="state">The state.</param>
+    public SimpleTextingResponseAsync( AsyncCallback callback, HttpContext context, Object state )
+    {
+        _callback = callback;
+        _context = context;
+        _state = state;
+        _completed = false;
+    }
+
+    /// <summary>
+    /// Starts the asynchronous work.
+    /// </summary>
+    public void StartAsyncWork()
+    {
+        ThreadPool.QueueUserWorkItem( ( workItemState ) =>
+        {
+            try
+            {
+                StartAsyncTask( workItemState );
+            }
+            catch ( Exception ex )
+            {
+                Rock.Model.ExceptionLogService.LogException( ex );
+                _context.Response.StatusCode = 500;
+                _completed = true;
+                _callback( this );
+            }
+        }, null );
+    }
+
+    /// <summary>
+    /// Starts the asynchronous task.
+    /// </summary>
+    /// <param name="workItemState">State of the work item.</param>
+    private void StartAsyncTask( Object workItemState )
+    {
+        var request = _context.Request;
+        var response = _context.Response;
+
+        response.ContentType = "text/json";
+
+        if ( request.HttpMethod != "POST" )
+        {
+            response.Write( "Invalid request type." );
+            response.StatusCode = ( int ) System.Net.HttpStatusCode.MethodNotAllowed;
+        }
+        else if ( !request.ContentType.Contains( "application/json" ) )
+        {
+            response.Write( "Invalid content type." );
+            response.StatusCode = ( int ) System.Net.HttpStatusCode.NotAcceptable;
+        }
+        else
+        {
+            var processedObj = ProcessJsonContent( request, response );
+            if ( processedObj != null )
+            {
+                switch ( processedObj.GetType().Name )
+                {
+                    case nameof( UnsubscribeReport ):
+                        // ignore Unsubscribe reports for now, perhaps if we add contact syncing or group sync with contact lists we will remove people here.
+                        break;
+                    case nameof( MessageReport ):
+                        var messageReport = ( MessageReport ) processedObj;
```

Comment:
Should probably check that the processedObj values are valid as well.

Reply:
I believe the checks in each sub-method are sufficient. I did look into newtonsoft validation method, they can validate json to be serialized but not the other direction.